### PR TITLE
Adapt to new SeqAn3 version

### PR DESCRIPTION
**Pull request by sarahet**
Still fails during build and I can't figure out the problem :confused:

### CODE REVIEW

**Comment by sarahet** [comment]
Now updated and works. Temporary solution as discussed: no bidirectional fm index.

File: src/mkindex_algo.hpp [hunk 1]
```diff
@@ -691,38 +691,55 @@ auto parseAndStoreTaxTree(std::vector<bool>          & taxIdIsPresent,
...
+        //     case AlphabetEnum::AMINO_ACID: // no reduction
+        //     {
+        //         index = TIndex{seqs};
+        //     } break;
+        //
+        //     default:
+        //     {
+        //         throw std::runtime_error{"This reduced alphabet is not handled correctly."};
+        //     } break;
+        // }
```
Comment (h-2) [inline src/mkindex_algo.hpp]: I think all of this including the next line can just be replaced with 
```cpp
index = TIndex{seqs};
```
Or not?

Comment (sarahet) [reply src/mkindex_algo.hpp]: You mean including the reduced version?

Comment (h-2) [reply src/mkindex_algo.hpp]: Something like this:
```cpp
if constexpr (std::same_as<seqan3::innermost_value_type_t<TStringSet>, seqan3::TRedAlph>)
{
     index = TIndex{seqs};
}
else
{
    auto redSeqs = seqs | seqan3::views::deep{seqan3::views::convert<TRedAlph>};
    index = TIndex{redSeqs};
}
```

So there are no more alphabet specific things.

Comment (sarahet) [reply src/mkindex_algo.hpp]: Ahhh I see. Yes I think that should work, looks valid.

Comment (sarahet) [reply src/mkindex_algo.hpp]: Works and looks better now :slightly_smiling_face:

**Pull request merged**